lib/csv/parser.rb

```diff
@@ -1021,18 +1030,19 @@ def parse_quotable_robust(&block)
           row << value
         elsif parse_row_end
           if row.empty? and value.nil?
-            emit_row([], &block) unless @skip_blanks
+            emit_row([], [], &block) unless @skip_blanks
           else
             row << value
```

Comment:
Do we need add an element to `quoted_field`s here?

```suggestion
            row << value
            quoted_fields << @quoted_column_value
```